### PR TITLE
Setting up PWA staging

### PR DESCRIPTION
**Pull request by Forrestdarabian**
# Description

Made a PWA component containing code for installing the Progressive Web App, also implemented the same code and applied it to the onClick of an install button to test it in card component

Provides an easier way for users to install Story Squad as a progressive web app

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] Tested Component Rendering

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by EricBanker12** [comment]
[[Parent] - Install button for PWA](https://trello.com/c/0OL2Keb2/53-parent-install-button-for-pwa)

File: src/app/components/common/pwa/installButton.component.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+const InstallButton: React.FC = () => {
+    const classes = useStyles({});
+    const [prompt, promptToInstall] = useAddToHomescreenPrompt();
+    const [isVisible, setVisibleState] = React.useState(false);
+
+    const hide = () => setVisibleState(false);
+
+    React.useEffect(() => {
+        console.log(prompt);
```
Comment (EricBanker12) [inline src/app/components/common/pwa/installButton.component.tsx]: remove console.log before merging

File: src/app/components/common/pwa/installButton.component.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        if (prompt) {
+            setVisibleState(true);
+        }
```
Comment (EricBanker12) [inline src/app/components/common/pwa/installButton.component.tsx]: needs to clear the prompt after installing or declining in OS

Add this after line 22
```
prompt.userChoice.then((platform, outcome) => {
    setVisibleState(false)
})
```

**Review by EricBanker12 — COMMENTED** [review]
Needs tests, and the prompt stays after installing.

**Review by EricBanker12 — COMMENTED** [review]
All tests passed, but manual testing showed that usePWA hook needs to be reworked into a context. It cannot capture the event if triggered at sign-in. It needs to listen from app, and send event to installButton.

**Review by EricBanker12 — APPROVED** [review]
Great work. All tests pass and manually testing on my mobile device gave the prompt as expected.

**Review by wSedlacek — APPROVED** [review]
Works on my end too. Good job

**Pull request merged**